src/sources/redis/list.rs

```diff
@@ -21,6 +22,7 @@ pub async fn watch(
     method: Method,
     decoder: codecs::Decoder,
     cx: SourceContext,
```

Comment:
None of the args here jump out as something that can be pulled out and reduced for the function, I'd be ok with a clippy ignore here.

Reply:
helper struct to contain?
I brought up the same issue when working on the `http_client` source and the result was introducing `GenericHttpClientInputs` .

Reply:
Sure that'd be fine too 😄

Reply:
ah, if the fields of the struct are pub, you can construct it without having to create a `new()` function.

For example: https://github.com/vectordotdev/vector/blob/f7f4b640719513ab594e324f64b493e315581091/src/sources/http_client/client.rs#L141-L150

Reply:
Allowing the lint would look like https://github.com/vectordotdev/vector/pull/14985/files#diff-0522e817840a91d701c913d3ecdd1f73ddf082e5d0d29f0f46a8ba7abb8e47d4R693

Reply:
Yeah I'm not going to die on this hill or anything- only bringing it up because this is the exact question I had a few months ago and it was discussed with the broader team and the solution linked in my prior comment was what others on the team suggested.

Reply:
I went ahead and implemented this in the newest set of changes. Thanks!